docs-src/app-dev-context/features.md

```diff
@@ -30,3 +30,4 @@ In this section you can find the following:
 - [How to set Activity timeouts & retries](#activity-timeouts)
 - [How to Heartbeat an Activity](#activity-heartbeats)
 - [How to Asynchronously complete an Activity](#asynchronous-activity-completion)
+- [How to register Namespaces](#namespaces)
```

Comment:
I am not sure this is how we want to tell people to register namespaces. Yes they _can_ programmatically register namespaces, and yes we may want to document this as a _possibility_, but I think we encourage use of `tctl` here.

Reply:
yeah. Updated the description to:
"Use the tctl command-line tool to [register Namespaces](/tctl/namespace/register).
You can also register your Namespaces programmatically with the `RegisterNamespace` API."

Reply:
:+1: So where does the `/go/how-to-register-namespace-in-go.md` file contents get included in the dev guide? Can we delete that file?

Reply:
Yeah; removed it from the docs/go so it should show only in the app dev guide context here: https://deploy-preview-1649--mystifying-fermi-1bc096.netlify.app/application-development/features?lang=go#namespaces

Reply:
I think programmatically registering a namespace should not be in the app dev guide. While Go unfortunately has a high level wrapper for this, other SDKs consider this a low-level service call like the dozen of others we don't document in the app dev guide.

But this is subjective. I would like to hear others' opinions on its value.

Reply:
I've seen a few questions come up about this on Slack and on community, and we point them to the APIs. I get what you're saying though;
If you look at it in the context of "this is a feature (Namespaces), and these are all the things you can do about it (register, update, set)", would the app dev guide inclusion make sense?

Reply:
I think it is fair to call out that Namespace creation is most commonly done from the command line tool -
But from a product coverage perspective, this makes things a lot more complete - and that matters a lot.

Reply:
I thought all SDKs expose public api (RegisterNamespaceRequest)? 
Yeah we had users ask us "how i register ns with code" in the past cause they dont want to use entry point scripts but if its in docs or not i guess then we should document the entire api? i really don't know

Reply:
>  But from a product coverage perspective, this makes things a lot more complete - and that matters a lot.

Then I have a lot more low-level API calls for you to document :-) List, update, describe, and delete/deprecate namespace for example. I couldn't imagine us having programmatic sections for all of those though arguably some of those are more useful than the one-time create.

Reply:
added high-level info for list, update, describe, and deprecate in concepts for managing Namespaces. Will update the app dev guide with examples for each in another PR.